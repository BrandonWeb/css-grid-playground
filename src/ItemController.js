import React from 'react';
import styled from 'styled-components';

const Button = styled.button`
  width: 25px;
  background: white;
  border: 0;
  text-align: center;
  cursor: pointer;
  &:nth-child(3) {
    margin: 0 0 0 5px;
  }
  &:last-child {
    margin: 0 5px 0 10px;
  }
`;
class ItemController extends React.Component {
  render() {
    return (
      <div>
        <Button onClick={this.props.onIncrease}>ðŸ‘†</Button>
        <Button onClick={this.props.onDecrease}>ðŸ‘‡</Button>
        <Button onClick={this.props.onReset}>ðŸ‘‹</Button>
        <Button onClick={this.props.onHideStyle}>ðŸ‘€</Button>
      </div>
    );
  }

}

export default ItemController;
