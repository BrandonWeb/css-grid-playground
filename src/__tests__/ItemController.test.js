import React from 'react';
import renderer from 'react-test-renderer';
import ItemController from '../ItemController';

it('renders correctly', () => {
  const tree = renderer.create(<ItemController />).toJSON();
  expect(tree).toMatchSnapshot();
});
