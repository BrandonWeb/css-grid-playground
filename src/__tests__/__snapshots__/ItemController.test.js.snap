exports[`test renders correctly 1`] = `
<div>
  <button
    className="czOWQi"
    onClick={null}>
    ğŸ‘†
  </button>
  <button
    className="czOWQi"
    onClick={null}>
    ğŸ‘‡
  </button>
  <button
    className="czOWQi"
    onClick={null}>
    ğŸ‘‹
  </button>
  <button
    className="czOWQi"
    onClick={null}>
    ğŸ‘€
  </button>
</div>
`;
